src/d/d_s_name.cpp

```diff
@@ -100,14 +105,91 @@ s32 dScnName_c::create() {
     if (rt == cPhs_COMPLEATE_e) {
         heap = JKRCreateExpHeap(0x68000, mDoExt_getGameHeap(), false);
         JUT_ASSERT(0x1c8, heap != 0);
-    }
 
-    oldHeap = mDoExt_setCurrentHeap(heap);
-    mArchive = new JKRMemArchive();
-    mArchive->mountFixed(dComIfG_getStageRes("Stage", 0x17), JKRMEMBREAK_FLAG_UNKNOWN0);
-    cloth_create();
-    buttonIconCreate();
-    dFs_c = new dFile_select_c();
+        oldHeap = mDoExt_setCurrentHeap(heap);
+        mArchive = new JKRMemArchive();
+        mArchive->mountFixed(dComIfG_getStageRes("Stage", 0x17), JKRMEMBREAK_FLAG_UNKNOWN0);
+        cloth_create();
+        buttonIconCreate();
+        dFs_c = new dFile_select_c();
+        JUT_ASSERT(470, dFs_c != 0);
+        dFs_c->field_0x0 = mArchive;
+        savePicDatabuf = new (0x20) u8[0x12000];
+        JUT_ASSERT(476, savePicDatabuf != 0);
+        if (fopAcM_GetName(this) == PROC_NAME_SCENE) {
```

Comment:
semantically I think `fopAcM_GetName` is supposed to be limited to actors while `fpcM_GetName` would be used for other processes, like scenes in this case. though it probably doesn't matter since they're functionally equivalent.